integration/findandmodify_compat_test.go

```diff
@@ -333,9 +333,20 @@ func TestFindAndModifyCompatUpdateSet(t *testing.T) {
 		},
 		"UpdateID": {
 			command: bson.D{
-				{"update", bson.D{{"$set", bson.D{{"_id", "non-existent"}}}}},
```

Comment:
Use `int32` instead of ``non-existent` because behaviour is different if collection contains `int23` or not.

Reply:
Isn't it a different case?

Reply:
What do you mean by that? 

When compat test runs on `Scalars` or `Int32` collection, it contains _id `int32` and this query does not have filter, so it does not return error. This was discovered just running compat test.
See also comment in the code 
> // when filter is not set and if collection contains ID to set, it does not error

When compat test runs on other collections, it doesn't contain `int32` so behaviour is the same as `non-existent` id and returns error.

Reply:
Ok, got it! Then why for `SetImmutableID` below there was `non-existent` used and not `int32`?

Reply:
Yes we can use `int32` or `non-existent`. FYI that test uses `ArrayDocuments` provider and the purpose of that test is check error message, so it's more obvious to use `non-existent` to ensuring test returns expected error message. For example, if that test changes provider from `ArrayDocuments` to `Int32` it will fail but using `non-existent` will pass test for any provider.

For compat test, we want to use input that discovers as much cases as possible, hence using int32.

Reply:
Wise idea :+1: